apiserver/facades/agent/instancemutater/instancemutater.go

```diff
@@ -31,15 +32,33 @@ type InstanceMutaterV2 interface {
 	WatchLXDProfileVerificationNeeded(ctx context.Context, args params.Entities) (params.NotifyWatchResults, error)
 }
 
+// MachineService defines the methods that the facade assumes from the Machine
+// service.
+type MachineService interface {
+	// GetMachineUUID returns the UUID of a machine identified by its name.
+	// It returns a MachineNotFound if the machine does not exist.
+	GetMachineUUID(ctx context.Context, name coremachine.Name) (string, error)
+
+	// LXDProfiles returns the names of the LXD profiles on the machine.
+	LXDProfiles(ctx context.Context, mUUID string) ([]string, error)
+
+	// SetLXDProfiles adds the list of LXD profile names to the lxd_profile table
+	// for the given machine.
+	// [machineerrors.MachineNotFound] will be returned if the machine does not
```

Comment:
I'd eschew the implementation detail regarding errors altogether. An interface can't guarantee those. The comments should be of a general nature.

Reply:
Yeah definitely agree, done.